examples/readonly-objects-and-fields/readonly_objects_and_fields.bal

```diff
@@ -0,0 +1,96 @@
+import ballerina/io;
+
+type Details record {|
+    int id;
+    string country;
+|};
+
+// An object with both read-only and non-read-only fields.
+type Employee object {
+    string department;
+    // `details` is a `readonly` field which cannot be updated once the value is created.
+    // `details` also expects an immutable value, and thus the effective contextually expected type for
+    // `details` is `Details & readonly`.
+    readonly Details details;
+
+    function init(string department, readonly & Details details) {
+        self.department = department;
+        self.details = details;
+    }
+};
+
+// An abstract object with non-readonly fields.
+type Controller abstract object {
+   int id;
+   string[] codes;
+
+   function getId() returns string;
+};
+
+// A non-abstract object with all read-only fields.
+type DefaultController object {
+    readonly int id;
+    readonly string[] codes;
+
+    function init(int id, readonly & string[] codes) {
+        self.id = id;
+        // The expected type for `codes` is `readonly & string[]`.
+        self.codes = codes;
+    }
+
+    function getId() returns string {
+        return string `Default: ${self.id}`;
+    }
+};
+
+// A non-abstract `readonly object`.
+type MainController readonly object {
+    int id;
+    string[] codes;
+
+    function init(int id, readonly & string[] codes) {
+        self.id = id;
+        // The expected type for `codes` is `readonly & string[]`.
+        self.codes = codes;
+    }
```

Comment:
Should we consider explaining this a bit more? Like, how we don't need to specify each field as readonly if the object is read only. I suppose why a plain int can be assigned to `id` is covered in the readonly BBE?

Reply:
Not repeat per se. Just to point and highlight the fields that they don't need to be marked readonly when the object itself is readonly.